src/ray/raylet/worker_pool.h

```diff
@@ -625,18 +626,51 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
       const PopWorkerStatus &status, bool *found /* output */,
       bool *worker_used /* output */, TaskID *task_id /* output */);
 
+  /// We manage all runtime env resources locally by the two methods:
+  /// `CreateRuntimeEnvIfNeeded` and `DeleteRuntimeEnvIfNeeded`.
+  ///
+  /// `CreateRuntimeEnvIfNeeded` means increasing the reference count for the runtime env
+  /// and `DeleteRuntimeEnvIfNeeded` means decreasing the reference count. We increase or
+  /// decrease runtime env reference in the cases below:
```

Comment:
Can you mention this happens from the agent now?

Reply:
actual ref counting

Reply:
ok